clang/test/SemaSYCL/device_has_ast.cpp

```diff
@@ -69,9 +70,34 @@ class KernelFunctor {
   }
 };
 
+// CHECK: CXXRecordDecl {{.*}} KernelFunctorAttr
+class KernelFunctorAttr {
+public:
+  // CHECK: CXXMethodDecl {{.*}} used operator() 'void () const'
+  // CHECK: SYCLDeviceHasAtt
+  // CHECK-NEXT: DeclRefExpr {{.*}} 'sycl::aspect' EnumConstant {{.*}} 'cpu' 'sycl::aspect'
+  [[sycl::device_has(cl::sycl::aspect::cpu)]] void operator()() const {}
```

Comment:
Should there also be a test for this attribute position?

```
void operator() [[sycl::device_has(cl::sycl::aspect::cpu)]] () const {}
```

Reply:
I don't think it is necessary because if it works for `[[sycl::device_has(cl::sycl::aspect::cpu)]] void operator()() const {}`, it should work for `void operator() [[sycl::device_has(cl::sycl::aspect::cpu)]] () const {}`. I can add it if you prefer though.